api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -0,0 +1,82 @@
+option java_multiple_files = true;
+
+// [#protodoc-title: HTTP Cache Filter]
+// HTTP Cache Filter :ref:`overview <config_http_filters_cache>`.
+// [#extension: envoy.extensions.filters.http.cache]
+
+// [#next-free-field: 6]
+message CacheConfig {
+  // [#not-implemented-hide:]
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
+
+    // If true, exclude the host from the cache key. Set to true if your origins' responses don't
+    // ever depend on host.
+    bool exclude_host = 2;
+
+    // If *query_parameters_included* is nonempty, only query parameters matched
+    // by one or more of its matchers are included in the cache key. Any other
+    // query params will not affect cache lookup.
+    repeated envoy.api.v2.route.QueryParameterMatcher query_parameters_included = 3;
+
+    // If *query_parameters_excluded* is nonempty, query parameters matched by one
+    // or more of its matchers are excluded from the cache key (even if also
+    // matched by *query_parameters_included*), and will not affect cache lookup.
+    repeated envoy.api.v2.route.QueryParameterMatcher query_parameters_excluded = 4;
+  }
+
+  // Config specific to the cache storage implementation.
+  google.protobuf.Any typed_config = 1;
+
+  // Name of cache implementation to use, as specified in the intended HttpCacheFactory
+  // implementation. Cache names should use reverse DNS format, though this is not enforced.
+  string name = 2;
+
+  // [#not-implemented-hide:]
+  // <TODO(toddmgreer) implement *vary* headers>
+  //
+  // List of allowed *Vary* headers.
+  //
+  // The *vary* response header holds a list of header names that affect the
+  // contents of a response, as described by
+  // https://httpwg.org/specs/rfc7234.html#caching.negotiated.responses.
+  //
+  // During insertion, *allowed_vary_headers* acts as a whitelist: if a
+  // response's *vary* header mentions any header names that aren't in
+  // *allowed_vary_headers*, that response will not be cached.
+  //
+  // During lookup, *allowed_vary_headers* controls what request headers will be
+  // sent to the cache storage implementation.
+  repeated type.matcher.v3alpha.StringMatcher allowed_vary_headers = 3;
+
+  // [#not-implemented-hide:]
+  // <TODO(toddmgreer) implement key customization>
+  //
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  KeyCreatorParams key_creator_params = 4;
+
+  // [#not-implemented-hide:]
+  // <TODO(toddmgreer) implement size limit>
+  //
+  // Max body size the cache filter will insert into a cache. 0 means unlimited (though the cache
+  // storage implementation may have its own limit beyond which it will reject insertions).
+  uint32 max_body_bytes = 5;
+}
```

Comment:
I think some of the issue you are encountering are because you are starting with a v3 proto rather than v2 and upgrading. I would like to support this flow properly in https://github.com/envoyproxy/envoy/issues/9769, but this is WiP. I recommend subscribing to that to track where we are going, since you are likely to need what is put together there.

Reply:
I reverted all my v3 changes, merged from upstream, and got everything back into a state that compiles and doesn't break proto_format.sh. It does still include v2, but it also has a v3 version.